source/fundamentals/connection/connection-options.txt

```diff
@@ -74,63 +92,95 @@ parameters of the connection URI to specify the behavior of the client.
        constraints in other ways, use a
        :ref:`custom SSLContext <tls-custom-sslContext>`.
 
+       | **Default**: ``false``
+
    * - **tlsAllowInvalidHostnames**
      - boolean
      - Specifies that the driver should allow invalid hostnames in the
        certificate for TLS connections. Supersedes
        **sslInvalidHostNameAllowed**.
 
+       | **Default**: ``false``
+
    * - **connectTimeoutMS**
      - integer
      - Specifies the maximum amount of time, in milliseconds, the Java
-       driver waits for a connection to open before timing out.
+       driver waits for a connection to open before timing out. A value of
+       ``0`` instructs the driver to never time out while waiting for a connection
+       to open.
+
+       | **Default**: ``10000`` (10 seconds)
 
    * - **socketTimeoutMS**
      - integer
      - Specifies the maximum amount of time, in milliseconds, the Java
        driver will wait to send or receive a request before timing out.
+       A value of ``0`` instructs the driver to never time out while waiting
+       to send or receive a request.
+
+       | **Default**: ``0`` (never times out)
```

Comment:
issue: I feel this is inconsistent with the other numerical defaults, and it is explicitly stated that 0 never times out (good addition for explanation!)

```suggestion
       | **Default**: ``0``
```